### PR TITLE
Convert discovery-* from integTest to [yaml | java]RestTest or internalClusterTest

### PR DESCRIPTION
**Pull request by jakelandis**
For OSS plugins that begin with `discovery-*`, the `integTest`
task is now a no-op and all of the tests are now executed via a test,
`yamlRestTest`, `javaRestTest`, or `internalClusterTest`.

related: #56841
related: #59444

### CODE REVIEW

File: plugins/discovery-azure-classic/build.gradle [hunk 1]
```diff
@@ -88,9 +90,10 @@ task createKey(type: LoggedExec) {
     '-keypass', 'keypass',
     '-storepass', 'keypass'
 }
-
+//no unit tests
+test.enabled = false
```
Comment (mark-vieira) [inline plugins/discovery-azure-classic/build.gradle:94]: This seems like one step forward and one step back. I assume this is also just to appease our testing conventions stuff? If so I feel like this should not be a failure.

File: plugins/discovery-ec2/qa/amazon-ec2/build.gradle [hunk 2]
```diff
@@ -66,19 +65,26 @@ integTest.enabled = false
  */
 ['KeyStore', 'EnvVariables', 'SystemProperties', 'ContainerCredentials', 'InstanceProfile'].forEach { action ->
   AntFixture fixture = tasks.create(name: "ec2Fixture${action}", type: AntFixture) {
-    dependsOn compileTestJava
-    env 'CLASSPATH', "${-> project.sourceSets.test.runtimeClasspath.asPath}"
+    dependsOn compileYamlRestTestJava
```
Comment (mark-vieira) [inline plugins/discovery-ec2/qa/amazon-ec2/build.gradle:68]: This should really be `dependsOn project.sourceSets.yamlRestTest.runtimeClasspath`. Just `compileYamlRestTestJava` doesn't include a) processing resources or b) upstream runtime dependencies so we risk a race condition here.

File: plugins/discovery-gce/qa/gce/build.gradle [hunk 3]
```diff
@@ -42,26 +40,26 @@ restResources {
 
 /** A task to start the GCEFixture which emulates a GCE service **/
 task gceFixture(type: AntFixture) {
-  dependsOn compileTestJava
-  env 'CLASSPATH', "${-> project.sourceSets.test.runtimeClasspath.asPath}"
+  dependsOn compileYamlRestTestJava
```
Comment (mark-vieira) [inline plugins/discovery-gce/qa/gce/build.gradle:43]: Same as above, we should depending on the source set runtime classpath here, not just the compile task.

**Review by mark-vieira — APPROVED** [review]
Couple minor comments, otherwise LGTM.

This also is another reminder that we need to get rid of these gross ant fixtures.

File: plugins/discovery-azure-classic/build.gradle (hunk 1 above)
Comment (jakelandis) [reply plugins/discovery-azure-classic/build.gradle:94]: yup.  However, it is a bit more rare to have no unit tests (at least until i start converting the qa projects).  For this one we will probably want to toggle the enable/disable the test task if the directory exists or not. I can look into that for a follow up PR.

Comment (mark-vieira) [reply plugins/discovery-azure-classic/build.gradle:94]: Yeah, we are going to have this all over the place for the qa projects that contain only rest tests.

File: plugins/discovery-gce/qa/gce/build.gradle (hunk 3 above)
Comment (jakelandis) [reply plugins/discovery-gce/qa/gce/build.gradle]: done https://github.com/elastic/elasticsearch/pull/60084/commits/01114f9eca06b6d834cf5c85bdc93bdc80faeb9f

File: plugins/discovery-ec2/qa/amazon-ec2/build.gradle (hunk 2 above)
Comment (jakelandis) [reply plugins/discovery-ec2/qa/amazon-ec2/build.gradle]: done https://github.com/elastic/elasticsearch/pull/60084/commits/01114f9eca06b6d834cf5c85bdc93bdc80faeb9f

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Comment by jakelandis** [comment]
@elasticmachine update branch

**Pull request merged**